test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -493,6 +493,68 @@ def skip_s3plugin_test(request):
             )
 
 
+@pytest.fixture(autouse=True)
+def skip_smdebug_test(
+    request,
+    processor,
+    ecr_image,
+):
+    if request.node.get_closest_marker("skip_smdebug_test"):
+        _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+        image_cuda_version = get_cuda_version_from_tag(ecr_image) if processor == "gpu" else ""
+        if (
+            Version(image_framework_version) in SpecifierSet("==2.0.1")
+            and Version(image_cuda_version.strip("cu")) == Version("121")
+        ) or Version(image_framework_version) in SpecifierSet(">=2.1"):
+            pytest.skip(
+                f"SM Profiler v1 is on path for deprecation, skipping this container with tag {image_framework_version}"
+            )
+
+
+@pytest.fixture(autouse=True)
+def skip_dgl_test(
+    request,
+    processor,
+    ecr_image,
+):
+    if request.node.get_closest_marker("skip_dgl_test"):
+        _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+        image_cuda_version = get_cuda_version_from_tag(ecr_image) if processor == "gpu" else ""
+        if (
+            Version(image_framework_version) in SpecifierSet("==2.0.1")
+            and Version(image_cuda_version.strip("cu")) == Version("121")
+        ) or Version(image_framework_version) in SpecifierSet(">=2.1"):
+            pytest.skip(
+                f"DGL binary is removed, skipping this container with tag {image_framework_version}"
+            )
+
+
+@pytest.fixture(autouse=True)
+def skip_smmodelparallel_test(
+    request,
+    processor,
+    ecr_image,
+):
+    if request.node.get_closest_marker("skip_smmodelparallel_test"):
+        _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+        image_cuda_version = get_cuda_version_from_tag(ecr_image) if processor == "gpu" else ""
+        if (
+            Version(image_framework_version) in SpecifierSet("==2.0.1")
+            and Version(image_cuda_version.strip("cu")) == Version("121")
+        ) or Version(image_framework_version) in SpecifierSet(">=2.1"):
+            pytest.skip(
+                f"SM Model Parallel team is maintaining their own Docker Container, skipping this container with tag {image_framework_version}"
+            )
+
+
+@pytest.fixture(autouse=True)
+def skip_p5_tests(instance_type, processor, ecr_image):
+    if "p5." in instance_type:
+        image_cuda_version = get_cuda_version_from_tag(ecr_image)
+        if processor != "gpu" or Version(image_cuda_version.strip("cu")) < Version("120"):
+            pytest.skip("Images using less than CUDA 12.0 doesn't support P5 EC2 instance.")
```

Comment:
nit: I know you just moved this function, but can change to "p5 instances require cu12 or higher"

Reply:
There are multiple skip_p5_tests definition in multiple conftest.py files. I'd rather not touch the function name in the PR.

Reply:
This doesn't require changing function name, just correcting the grammar in the skip condition. However, it is a nit, so feel free to resolve